cpu/stm32/periph/spi.c

```diff
@@ -134,12 +134,20 @@ void spi_init_pins(spi_t bus)
     gpio_init_af(spi_config[bus].mosi_pin, GPIO_AF_OUT_PP);
     gpio_init(spi_config[bus].miso_pin, GPIO_IN);
 #else
-    gpio_init(spi_config[bus].mosi_pin, GPIO_OUT);
-    gpio_init(spi_config[bus].miso_pin, GPIO_IN);
-    gpio_init(spi_config[bus].sclk_pin, GPIO_OUT);
-    gpio_init_af(spi_config[bus].mosi_pin, spi_config[bus].mosi_af);
-    gpio_init_af(spi_config[bus].miso_pin, spi_config[bus].miso_af);
-    gpio_init_af(spi_config[bus].sclk_pin, spi_config[bus].sclk_af);
+    if (spi_config[bus].mosi_pin != GPIO_UNDEF) {
```

Comment:
```suggestion
    if (gpio_is_valid(spi_config[bus].mosi_pin)) {
```

should be more future-proof

Reply:
Makes sense